src/components/search/SearchDirective.js

```diff
@@ -1,756 +1,259 @@
-              scope.timeStamps = timeStamps;
             });
+            unregisterLayers();
+          });
+          unregister();
+        });
+      }
 
-            // We have to create a small workaround to get the
-            // a suggestionsRendered event that includes the
-            // dataset
-            var viewDropDown = taElt.data('ttView').dropdownView;
-            var renderSuggestions = viewDropDown.renderSuggestions;
-            var renderByDataset = function(context, dataset, suggestions,
-                invokeApply) {
-              if (dataset.name === 'locations') {
-                hasLocationResults = (suggestions.length !== 0);
-              } else if (dataset.name === 'featuresearch') {
-                hasFeatureResults = (suggestions.length !== 0);
-              } else if (dataset.name === 'layers') {
-                // hasLayerResults is used to control
-                // the display of the footer
-                $timeout(function() {
-                  scope.hasLayerResults = (suggestions.length !== 0);
-                });
-              }
-              renderSuggestions.apply(context, [dataset, suggestions]);
+    }
+  );
 
-              if (invokeApply !== false) {
-                scope.$apply(function() {
-                  context.trigger('gaSuggestionsRendered', dataset);
-                });
-              } else {
-                context.trigger('gaSuggestionsRendered', dataset);
-              }
-            };
-            var immediate = (!scope.options.renderWait);
-            var renderLocations = gaDebounce.debounce(renderByDataset,
-                scope.options.renderWait, immediate);
-            var renderFeatures = gaDebounce.debounce(renderByDataset,
-                scope.options.renderWait, immediate);
-            var renderLayers = gaDebounce.debounce(renderByDataset,
-                scope.options.renderWait, immediate);
-            viewDropDown.renderSuggestions = function(dataset, suggestions,
-                invokeApply) {
-              // To avoid slow typing search we need to debounce the trigger of
-              // gaSuggestionRendered events
-              if (dataset.name === 'locations') {
-                renderLocations(this, dataset, suggestions, invokeApply);
-              } else if (dataset.name === 'featuresearch') {
-                renderFeatures(this, dataset, suggestions, invokeApply);
-              } else if (dataset.name === 'layers') {
-                renderLayers(this, dataset, suggestions, invokeApply);
-              }
-            };
-
-            // Adapt dynamically the list height according to the
-            // number of lists
-            scope.nbOfSuggestionsLists = {
-              'ga-search-1': false,
-              'ga-search-2': false,
-              'ga-search-3': false
-            };
+  module.directive('gaSearch',
+      function($timeout) {
+        return {
+          restrict: 'A',
+          replace: true,
```

Comment:
Replace is drepecated : https://code.angularjs.org/1.3.0-beta.11/docs/api/ng/service/$compile#-replace-deprecated-will-be-removed-in-next-major-release-